### PR TITLE
fix: linux launcher breaks when there are spaces in the directory path

### LINKED ISSUE #4791: [Bug] Linux launcher breaks when there are spaces in the directory path
### Description of the issue

Really dumb issue. Downloading version 1.1.747 for Linux ([here](https://github.com/Ryujinx/release-channel-master/releases/download/1.1.747/ryujinx-1.1.747-linux_x64.tar.gz), and running `./Ryujinx.sh` after extracted, you're met with an error if the working directory path contains a space. Adding some quotes fixes this.

### Reproduction steps

1. Use spaces in your folder names
2. Attempt to run ./Ryujinx.sh
3. Errors!

This can be fixed by changing this:
`SCRIPT_DIR=$(dirname $(realpath $0))`
To this:
`SCRIPT_DIR=$(dirname "$(realpath $0)")`

### Log file

`azzy@pop-desktop:~/Downloads/Debs/Not Debs/ryujinx-1.1.747$ ./Ryujinx.sh
env: ‘/home/azzy/Downloads/Debs\nDebs/ryujinx-1.1.747/Ryujinx’: No such file or directory
`

After adding the quotes as described above:

`azzy@pop-desktop:~/Downloads/Debs/Not Debs/ryujinx-1.1.747$ ./Ryujinx.sh
00:00:00.006 |I| Configuration LogValueChange: EnableFileLog set to: True
00:00:00.045 |I| Configuration LogValueChange: BackendThreading set to: Auto
00:00:00.046 |I| Configuration LogValueChange: ResScale set to: 1
00:00:00.049 |I| Configuration LogValueChange: ResScaleCustom set to: 1
00:00:00.049 |I| Configuration LogValueChange: MaxAnisotropy set to: -1
00:00:00.049 |I| Configuration LogValueChange: AspectRatio set to: Fixed16x9
00:00:00.050 |I| Configuration LogValueChange: GraphicsBackend set to: OpenGl
00:00:00.051 |I| Configuration LogValueChange: PreferredGpu set to: 
00:00:00.053 |I| Configuration LogValueChange: EnableDockedMode set to: True
00:00:00.059 |I| Configuration LogValueChange: EnableVsync set to: True
00:00:00.060 |I| Configuration LogValueChange: EnableShaderCache set to: True
00:00:00.060 |I| Configuration LogValueChange: EnableTextureRecompression set to: False
00:00:00.060 |I| Configuration LogValueChange: EnableMacroHLE set to: True
00:00:00.060 |I| Configuration LogValueChange: AntiAliasing set to: None
00:00:00.061 |I| Configuration LogValueChange: ScalingFilter set to: Bilinear
00:00:00.061 |I| Configuration LogValueChange: ScalingFilterLevel set to: 80
00:00:00.061 |I| Configuration LogValueChange: EnablePtc set to: True
00:00:00.061 |I| Configuration LogValueChange: EnableInternetAccess set to: False
00:00:00.061 |I| Configuration LogValueChange: EnableFsIntegrityChecks set to: True
00:00:00.061 |I| Configuration LogValueChange: FsGlobalAccessLogMode set to: 0
00:00:00.061 |I| Configuration LogValueChange: AudioBackend set to: SDL2
00:00:00.061 |I| Configuration LogValueChange: AudioVolume set to: 1
00:00:00.062 |I| Configuration LogValueChange: MemoryManagerMode set to: HostMappedUnsafe
00:00:00.062 |I| Configuration LogValueChange: ExpandRam set to: False
00:00:00.062 |I| Configuration LogValueChange: IgnoreMissingServices set to: False
00:00:00.062 |I| Configuration LogValueChange: UseHypervisor set to: True
00:00:00.255 |N| Application PrintSystemInfo: Ryujinx Version: 1.1.747
00:00:00.263 |N| Application Print: Operating System: Linux 6.2.6-76060206-generic #202303130630~1681329778~22.04~d824cd4 SMP PREEMPT_DYNAMIC Wed A (X64)
00:00:00.263 |N| Application Print: CPU: AMD Ryzen 7 5700X 8-Core Processor ; 16 logical
00:00:00.264 |N| Application Print: RAM: Total 15908 MiB ; Available 8953 MiB
00:00:00.266 |N| Application PrintSystemInfo: Logs Enabled: Info, Warning, Error, Guest, Stub
00:00:00.266 |N| Application PrintSystemInfo: Launch Mode: UserProfile
00:00:02.677 |W| Application ApplyTheme: The "custom_theme_path" section in "Config.json" contains an invalid path: "Ryujinx.Common.ReactiveObject`1[System.String]".
00:00:05.775 |I| Configuration LogValueChange: GraphicsBackend set to: Vulkan`

And it launches with no issue.

### OS

Pop!_OS 22.04

### Ryujinx version

1.1.747

### Game version

_No response_

### CPU

_No response_

### GPU

_No response_

### RAM

_No response_

### List of applied mods

_No response_

### Additional context?

_No response_

### PR DESCRIPTION
**Pull request by 0xHexE**
Fixes #4791

### CODE REVIEW

**Review by marysaka — APPROVED** [review]

**Review by TSRBerry — COMMENTED** [review]
Should there also be quotes around $0?

**Comment by marysaka** [comment]
> Should there also be quotes around $0?

That's actually true!

File: distribution/linux/Ryujinx.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-SCRIPT_DIR=$(dirname $(realpath $0))
+SCRIPT_DIR=$(dirname "$(realpath $0)")
```
Comment (marysaka) [inline distribution/linux/Ryujinx.sh:3]: Need quotes around $0

Comment (TSRBerry) [reply distribution/linux/Ryujinx.sh]: I just added them, since we don't really need to delay this just for a small change.
Does that look fine to you?

**Review by TSRBerry — APPROVED** [review]
Lgtm, thank you!

**Pull request merged**